Init Backend Server

mike10911 opened the pull request:
### Change list:
- Created /backend where the cabin and judging algorithms live
- init a node server in /backend
- Setup an express server in app.ts (entry point)

Note:
Still deciding if we are going to deploy with render.com or Vercel

surbhi-gulati commented on backend/app.ts:
please eslint the file

surbhi-gulati commented on backend/package.json:
```diff
@@ -0,0 +1,21 @@
+  "name": "backend",
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.tnpms",
```
typo? assuming should be app.ts

surbhi-gulati commented on backend/app.ts:
```diff
@@ -0,0 +1,20 @@
+
+app.use(
+    cors({
+        origin: 'http://localhost:3000',
```
choose a different port, as this is conventionally what frontend runs on when we are operating locally

surbhi-gulati reviewed (changes requested):
Cannot get the app running - can you list steps to reproduce? I just went to backend directory and put in `nodemon app.ts` but no avail :(

mike10911 commented:
New Changes:
- Added eslint config files
- Updated port from 3000-->4000
- Fix package.json typoes
- Add new tsconfig.json
- Added prettier config file

mike10911 commented:
### How to Run New Internal Tools Central Backend Server:

1. cd `backend` directory
2. run `yarn install` to install dependencies
3. run `yarn dev` to run the express server on port 4000

surbhi-gulati reviewed (approved):
See slack. Thanks for setting this up!!

NelsonDong41 reviewed (approved):
Looks good, console log is properly showing
